velox/expression/Expr.cpp

```diff
@@ -1439,6 +1439,29 @@ void Expr::applyFunction(
 
   vectorFunction_->apply(rows, inputValues_, type(), context, result);
 
+  if (!result) {
```

Comment:
wondering if we should instead enforce a constraint (by throwing or adding a DCHECK) that a vectorFunction should always set result, that would instead put onus on the UDF and avoid an extra check like this.

Reply:
Thanks for the review!

1) I don't think a DCHECK is sufficient, making it just a DCHECK could lead to crashes/undefined behavior at runtime.  One challenge with DCHECKs checking UDF implementation is there's no guarantee that the user executed Velox in debug mode with their UDF to test it, e.g. they may have just written some unit tests that target just the UDF implementation.
2) Even with the DCHECK and testing it seems easy to miss a case where, say you have the implementation of the Vector function wrapped in a try that on error sets the error pointers for all rows (which all Vector functions probably should have, so maybe we should be adding that to, but that's another conversation), but forgets to set the result.  If that exception is rare or in an untested case it could easily slip through.
3) This seems like a really cheap check compared to the rest of the expression evaluation code, I could even add an UNLIKELY hint for the compiler to improve performance if that's a concern.

Reply:
Thanks Kevin for the explanation. This looks good.